src/GraphQL/Builders/ConnectionBuilder.cs

```diff
@@ -194,7 +199,7 @@ public void ResolveAsync(Func<ResolveConnectionContext<TSourceType>, Task<object
 
         private void CheckForErrors(ResolveConnectionContext<TSourceType> args)
         {
-            if (args.First.HasValue && args.Last.HasValue)
+            if (args.First.HasValue && args.Last.HasValue && !_allowBidirectionalSlicing)
```

Comment:
@sungam3r seems the spec allows this? so we should just remove this arg check?

Reply:
I must admit that I still can’t find the time to read the specification and check this PR. I will try to do this ASAP.

Reply:
https://relay.dev/graphql/connections.htm#sec-Pagination-algorithm

Reply:
> **Pagination Algorithm:**
>
> To determine what edges to return, the connection evaluates the before and after cursors to filter the edges, then evaluates first to slice the edges, then last to slice the edges.
>
> _Note: Including a value for both first and last is strongly discouraged, as it is likely to lead to confusing queries and results. The PageInfo section goes into more detail here._

Reply:
The actual implementation isn't within the GraphQL library.  If we remove the arg check (which really should have been an `ExecutionError`) it will be up to the users to write code that accurately returns results.  See `GraphQL.StarWars.Extensions.ResolveFieldContextExtensions.GetPagedResults(...)`.

The original suggestion here might be best: to only allow "bidirectional slicing" when configured to do so.